### PR TITLE
add a hubot docker image

### PR DESCRIPTION
**Pull request by reaperhulk**
This requires a token passed as an environment variable to work properly. We'll need to figure out how we want to provide that.

### CODE REVIEW

**Comment by reaperhulk** [comment]
This is now r4r. Tested by adding `HUBOT_GITHUB_TOKEN` via `docker secret create` and then doing  `docker service  create --name="hubot" --secret="HUBOT_GITHUB_TOKEN" pyca/hubot`. The latter needs to be added to ansible after this is merged and we need to decide how to document the secret creation requirement...

File: hubot/data/Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: bin/hubot -a irc
```
Comment (alex) [inline hubot/data/Procfile]: What uses this?

File: hubot/data/external-scripts.json [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[
```
Comment (alex) [inline hubot/data/external-scripts.json]: What uses this?

**Comment by alex** [comment]
Actually, before we merge this, please create a `CREDENTIALS.md` with useful information.

**Comment by reaperhulk** [comment]
Information about what the credentials need to be? Do you want that to live in the hubot dir?

**Comment by alex** [comment]
No, in teh root of the repo please.

File: hubot/data/external-scripts.json (hunk 2 above)
Comment (reaperhulk) [inline hubot/data/external-scripts.json]: This is generated by yo script that makes hubot configs, but I don't know if it's actually required (similarly, not sure about procfile). I'll experiment and see quickly.

File: hubot/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+RUN mkdir -p /home/hubot/.config/configstore
+RUN mkdir /home/hubot/bot
+RUN chown -R hubot:hubot /home/hubot
+
+USER hubot
+WORKDIR /home/hubot/bot
+
+COPY ["data/package.json", "/home/hubot/bot/"]
+COPY ["data/scripts/", "/home/hubot/bot/scripts"]
+COPY ["data/bin/", "/home/hubot/bot/bin"]
```
Comment (alex) [inline hubot/Dockerfile]: These lines don't need the array syntax, do they? `COPY data/bin/ /home/hubot/bot/bin` works right?

File: hubot/data/bin/hubot [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+set -e
+
+npm install
```
Comment (alex) [inline hubot/data/bin/hubot]: Why does running `hubot` trigger an `npm install`?

Comment (reaperhulk) [inline hubot/data/bin/hubot]: No idea.  the `yo` script that creates a hubot does this. We could potentially remove it.

File: hubot/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,30 @@
+FROM node:7.9-slim
+
+RUN apt-get -qq update
+RUN apt-get -qq -y upgrade
+RUN apt-get install -qq -y runit
+
+ENV HUBOT_IRC_SERVER irc.freenode.net
+ENV HUBOT_IRC_ROOMS #cryptography-dev
```
Comment (alex) [inline hubot/Dockerfile]: Does this not need to be quoted?

Comment (reaperhulk) [inline hubot/Dockerfile]: "The ENV instruction has two forms. The first form, ENV <key> <value>, will set a single variable to a value. The entire string after the first space will be treated as the <value> - including characters such as spaces and quotes."

**Pull request merged**